server/src/main/java/org/opensearch/index/MergeSchedulerConfig.java

```diff
@@ -130,19 +138,68 @@ public final class MergeSchedulerConfig {
         Property.NodeScope
     );
 
+    private final String indexName;
+    private IndexSettings indexSettings;
     private volatile boolean autoThrottle;
     private volatile int maxThreadCount;
     private volatile int maxMergeCount;
     private volatile double maxForceMergeMBPerSec;
+    private static volatile Boolean cachedAutoThrottleEnabledDefault;
+    private static volatile Integer cachedMaxThreadCountDefault;
+    private static volatile Integer cachedMaxMergeCountDefault;
 
     MergeSchedulerConfig(IndexSettings indexSettings) {
-        int maxThread = indexSettings.getValue(MAX_THREAD_COUNT_SETTING);
-        int maxMerge = indexSettings.getValue(MAX_MERGE_COUNT_SETTING);
-        setMaxThreadAndMergeCount(maxThread, maxMerge);
-        this.autoThrottle = indexSettings.getValue(AUTO_THROTTLE_SETTING);
+        this.indexSettings = indexSettings;
+        indexName = indexSettings.getIndex().getName();
+        initMergeConfigs(indexSettings);
         updateMaxForceMergeMBPerSec(indexSettings);
     }
 
+    public synchronized void setDefaultMaxThreadAndMergeCount(int threadCount, int mergeCount, boolean overrideExistingConfigs) {
+        cachedMaxThreadCountDefault = threadCount;
+        cachedMaxMergeCountDefault = mergeCount;
+        if (overrideExistingConfigs == true) {
+            setMaxThreadAndMergeCount(threadCount, mergeCount);
+        }
+    }
+
+    public synchronized void setDefaultAutoThrottleEnabled(boolean enabled, boolean overrideExistingConfigs) {
+        cachedAutoThrottleEnabledDefault = enabled;
+        if (overrideExistingConfigs == true) {
+            setAutoThrottle(enabled);
+        }
+    }
+
+    private void initMergeConfigs(IndexSettings indexSettings) {
+        Settings settings = indexSettings.getSettings();
+        boolean useCachedClusterDefaults = MAX_THREAD_COUNT_SETTING.exists(settings) == false
+            && MAX_MERGE_COUNT_SETTING.exists(settings) == false;
```

Comment:
Couple of things:
1. move the conditions the conditions `cachedMaxThreadCountDefault != null` and `cachedMaxMergeCountDefault != null` here as `<existing_condition> && <new_condition>`?
2. can cachedMaxThreadCountDefault and cachedMaxMergeCountDefault be null in any circumstances? If so, lets remove it and add an assertion?